HyphaApp/hypha
# Improve testing and development experience

Fixes N/A

- Add `make test` command to lint, test, and report coverage
- Add pytest runner 
  - It automatically set up to use the correct Django settings module
  - Runs tests on all the available CPUs, I was able to run all the tests in total=~30sec using the `make test` command.
  - It provides a cleaner error report, compared to DjangoTestRunner
- Coverage
  - Displayed the report on the terminal after the test finish
  - Disable reporting of coverage for test and migration files
  - Enable coverage reporting for the template files
  - “make test” generates HTML reports to quickly check the coverage of each file
- Update CircleCI to use `pytest` command and CircleCI parallelism=4 (reduction in test time by ~2-3mins from 8-9min to ~5min)
